src/off-chain/historical-data/cowEndpointSurplus.py

```diff
@@ -0,0 +1,258 @@
+        endpoint_url = f"https://api.cow.fi/mainnet/api/v1/solver_competition/by_tx_hash/{settlement_hash}"
+        json_competition_data = requests.get(endpoint_url)
+
+        if json_competition_data.status_code == 200:
+            competition_data = json.loads(json_competition_data.text)
+            winning_solution = competition_data["solutions"][-1]
+            winning_solver = winning_solution["solver"]
+            winning_orders = winning_solution["orders"]
+
+            for individual_win_order in winning_orders:
+                self.solver_dict[winning_solver][0] += 1
+                self.total_orders = self.total_orders + 1
+                individual_win_order_id = individual_win_order["id"]
+                order_data_url = f"https://api.cow.fi/mainnet/api/v1/orders/{individual_win_order_id}"
+                json_order = requests.get(order_data_url)
+                if json_order.status_code == 200:
+                    individual_order_data = json.loads(json_order.text)
+                    if individual_order_data["isLiquidityOrder"] is False:
+                        # printing settlement_hash as a debug test
+                        print(settlement_hash)
+                        surplus_deviation_dict = {}
+                        soln_count = 0
+                        for soln in competition_data["solutions"]:
+                            for order in soln["orders"]:
+                                if individual_win_order_id == order["id"]:
+                                    (
+                                        diff_surplus,
+                                        percent_deviation,
+                                        surplus_token,
+                                    ) = self.get_surplus_difference(
+                                        individual_order_data, soln, order
+                                    )
+                                    surplus_eth = (
+                                        int(
+                                            competition_data["auction"]["prices"][
+                                                surplus_token
+                                            ]
+                                        )
+                                        / (pow(10, 18))
+                                    ) * (diff_surplus / pow(10, 18))
+                                    surplus_deviation_dict[soln_count] = (
+                                        surplus_eth,
+                                        percent_deviation,
+                                    )
+                            soln_count += 1
+                        self.print_function(
+                            surplus_deviation_dict,
+                            settlement_hash,
+                            individual_win_order_id,
+                            competition_data,
+                        )
+        else:
+            print("not able to fetch data from competition endpoint.")
```

Comment:
`get_order_surplus` is too long and doing multiple things. We should split it. 

For example, there is no way to calculate the surplus based on the order_id. For this, we should create a function that does only one thing: calculates the surplus based on the minimum necessary order data. It should be a standalone function (not part of any class) and easily unit tested.

Then, we call this function from other functions...

Reply:
_For this, we should create a function that does only one thing: calculates the surplus based on the minimum necessary order data._
Isn't this already get_surplus_difference, which uses minimum necessary data?